File: stacks/laravel/app/Domain/Greeting.php
<?php

declare(strict_types=1);

namespace App\Domain;

use InvalidArgumentException;

final class Greeting
{
    public static function for(string $name): string
    {
        if ($name === '') {
            throw new InvalidArgumentException('name is required');
        }

        return "Hello, {$name}.";
    }
}

File: stacks/go/src/core/greeting.go
// Package core holds pure domain logic with no I/O dependencies.
package core

import "fmt"

// Greet returns a greeting for name. It returns an error if name is empty.
func Greet(name string) (string, error) {
	if name == "" {
		return "", fmt.Errorf("name is required")
	}
	return fmt.Sprintf("Hello, %s.", name), nil
}
